client/src/pages/Prospects.js

```diff
@@ -120,97 +180,123 @@ const Prospects = () => {
+      handleSelectedProspects(newSelecteds);
+      return;
+    }
+    handleSelectedProspects([]);
+  };
 
-  const actionType = 'Add to Campaign'
+  //handle select one row on DataTable.js
+  const handleClickOnRow = (event, id) => {
+    const selectedIndex = selectedProspects.indexOf(id);
+    let newSelected = [];
+
+    if (selectedIndex === -1) {
+      newSelected = newSelected.concat(selectedProspects, id);
+    } else if (selectedIndex === 0) {
+      newSelected = newSelected.concat(selectedProspects.slice(1));
+    } else if (selectedIndex === selectedProspects.length - 1) {
+      newSelected = newSelected.concat(selectedProspects.slice(0, -1));
+    } else if (selectedIndex > 0) {
+      newSelected = newSelected.concat(
+        selectedProspects.slice(0, selectedIndex),
+        selectedProspects.slice(selectedIndex + 1),
+      );
+    }
+    handleSelectedProspects(newSelected);
+  };
 
   const propsForDialog = {
     actionType,
+    dataList: listOfCampaigns,
+    setValue: setCampaignId,
+    currentValue: campaignId,
     dialog,
-    campaigns,
-    setCampaignId,
-    campaignId,
-    handleCloseDialogAndSaveProspects
+    handleCloseDialogAndSaveProspects,
+    handleDialog
+  }
+
+  const propsForDataTable = {
+    data,
+    handleClickOnAllRows,
+    handleClickOnRow,
+    selectedProspects,
   }
 
   return (
     <Fragment>
       <NavBar />
-      <div>
-        <Box
-          className={classes.featuresContainer}
-          display="flex">
-          <Box flexGrow={1}>
-            <Typography variant="h5"> Prospects </Typography>
-          </Box>
-          <Box flexGrow={1}>
-          {prospects.length > 0 &&
-            <CustomizedButton
-              onClick={() => handleDialog(true)}>
-              {actionType}
-            </CustomizedButton>}
-          </Box>
-            {dialog === true &&
-              <CustomizedDialog
-              propsForDialog={propsForDialog}
-              />}
-          <Box className={classes.icon}>
-            <FlashOnIcon fontSize="small" style={{color: "grey"}} />
-          </Box>
-          <Box className={classes.icon}>
-            <MailIcon fontSize="small" style={{color: "grey"}} />
-          </Box>
-          <Box pl={2}>
-            <OutlinedButton className={classes.importButton}> Imports </OutlinedButton>
+      <Grid container>
+        <Grid item xs={2} id='sidePanel'>
+          <Box>
+            <SidePanel data={data} filterTerms={filterTerms} handleSearch={handleSearch}> </SidePanel>
           </Box>
-          <Box pl={1}>
-            <CustomizedButton 
-              className={classes.newProspectButton}>
-              Add New Prospect
-              <div className={classes.seperationLine}></div>
-              <div className={classes.arrow} >
-                <ArrowDropDownIcon fontSize="small" style={{color: "white"}} pt={3} />
-              </div>
-            </CustomizedButton>
+        </Grid>
+        <Grid item xs={10}>
```

Comment:
The side panel isn't really responsive right now, it would be cool if you could have the side panel be stacked on top of the page below the nav panel on smaller screens.